app/models/post.rb

```diff
@@ -1,5 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :user, class_name: 'User', foreign_key: 'user_id'
   has_many :comments
   has_many :likes
+
+  def update_post_counter
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
- [ ] The `#update_post_counter` method should be called every time a new post has been saved to database, you can perform that feature using `after_save` , to address that issue 👍  then you should make the `update_post_counter` method private because it will not be accessed in the class 😄 

```ruby
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

- [ ] [OPTIONAL] Instead of using `update` it is very appropriated to  used `increment` instead, Kindly fix that

Reply:
Solved :+1: